tangchengjin/demo.py

```diff
@@ -2,37 +2,63 @@
 from random import choice
 name = input("Input file name: ")  ## optional: loop
 ## 1.1. try.. except check file exists
+flag = 0
+while flag == 0 :
+    try :
+        fhand = open(name)
+        print(fhand)
```

Comment:
We usually only want to include code that _are prone to_ throwing an error in a `try` block. Here, only the line `fhand = open(name)` may throw an error, so it's of better style to put other statements outside of the `try` block.